services/nrfradio.md

```diff
@@ -0,0 +1,36 @@
+# NRF Radio
+
+    identifier: 0x1ac986cf
+
+## Registers
+
+    rw enabled: bool @ intensity
+
+Turns on/off the radio antenna.
+
+    rw group: u8 @ 0x80
+
+Group used to filter packets
+
+    rw transmission_power = 6: u8 { min_value=1, max_value=7 } @ 0x81
+
+Antenna power to increase or decrease range.
+
+## Commands
+
+    command send @ 0x80 {
+        data: bytes { max_bytes: 18 }
```

Comment:
I don't think we should limit this to what micro:bit v1 can do; the max len of packets is in the ~250 bytes range.

There is some `S0` and `S1` fields that I don't understand. Also I think that `group` is a codal concept (maybe it's just the first byte of payload?).

@jamesadevine should be able to say more.

Reply:
I was thinking of explicately support a v1-backward compatible radio protocol. Maybe we can have something more generic as another service.
The purpose of this service would be bridge JACDAC buses over radio too.

Reply:
The radio is restricted in software to communicate 32 bytes max (I think).

The max data length of a radio packet can be 255 bytes, not including metadata.

My big question is what is the purpose behind this service? bit:radio is a lossy protocol, it has no reliability guarantees and collisions between devices would be likely if running jacdac on top (if two devices communicate/overlap packets then both packets get lost).

If this service is simply to send and receive bit:radio packets with no reliability guarantees, then fine, great. But if you're envisaging jacdac over radio it would require a mesh protocol with TDMA management (what I did as part of my thesis) for reliability.

Reply:
> If this service is simply to send and receive bit:radio packets with no reliability guarantees, then fine, great.
Yes that's the goal. Imagine a cheap module with nrf51 that exposes sensor via radio. It's another way to look at JACDAC.

Reply:
I think a basic comms channel between bit:radio and jacdac sounds good. To run jacdac on a properietry radio, we would have to do something radically different (i.e. think about wireless in the same way as we did with single wire serial).

Remember packet payload is capped at 32 bytes!! :smile:

Reply:
Not even that. We reserve some bytes in MakeCode for time/serial etc...

Reply:
should we replicate makecode restrictions in this service?

Reply:
Yeah, hence the 18bytes. Maybe we need the other variants, send string, send number, ...